test/stdlib/Ractor_test.rb

```diff
@@ -0,0 +1,173 @@
+require_relative "test_helper"
+
+if RUBY_VERSION < '3'
+  warn 'Ractor is not available on Ruby 2🐫 Skip the test'
+  return
+end
+
+class RactorSingletonTest < Test::Unit::TestCase
+  include TypeAssertions
+
+  testing "singleton(::Ractor)"
+
+  def test_count
+    assert_send_type "() -> Integer",
+                     Ractor, :count
+  end
+
+  def test_current
+    assert_send_type "() -> Ractor",
+                     Ractor, :current
+  end
+
+  def test_main
+    assert_send_type "() -> Ractor",
+                     Ractor, :main
+  end
+
+  def test_make_shareable
+    assert_send_type "(String) -> String",
+                     Ractor, :make_shareable, 'foo'
+    assert_send_type "(String, copy: true) -> String",
+                     Ractor, :make_shareable, 'foo', copy: true
+    assert_send_type "(String, copy: false) -> String",
+                     Ractor, :make_shareable, 'foo', copy: false
+    assert_send_type "(String, copy: nil) -> String",
+                     Ractor, :make_shareable, 'foo', copy: nil
+  end
+
+  def test_new
+    # TODO: it raises an error because the proc is not isolated
+    # assert_send_type '() { () -> untyped } -> Ractor',
+    #                  Ractor, :new do end
+  end
+
+  def test_receive
+    Ractor.current.send 42
+    assert_send_type "() -> Integer",
+                     Ractor, :receive
+  end
+
+  def test_recv
+    Ractor.current.send 42
+    assert_send_type "() -> Integer",
+                     Ractor, :recv
+  end
+
+  def test_receive_if
+    Ractor.current.send 42
+    assert_send_type "() { (Integer) -> bool } -> Integer",
+                     Ractor, :receive_if do |n| n == 42 end
+  end
+
+  def test_select
+    r1 = Ractor.new { loop { Ractor.yield 42 } }
```

Comment:
This test code leaks Ractor processes in some places. The leaks don't have a large impact because this test code is short-lived code.